### PR TITLE
Fix method implementation

### PR DESCRIPTION
**Pull request by AnikanovD**
https://github.com/composer/composer/commit/623c0dcda7a29a06d21e9736527b05ca436d0f4b

In method _ComposerRepository::whatProvides()_ was added a new parameter. 
After _composer self-update_, it began to throw an exceptions
```
[ReflectionException]                                                   
Class Fxp\Composer\AssetPlugin\Repository\NpmRepository does not exist  
                                                                                                                                                                                                              
[ErrorException]                                                                                                                                                                                            
Declaration of Fxp\Composer\AssetPlugin\Repository\AbstractAssetsRepository::whatProvides() should be compatible with Composer\Repository\ComposerRepository::whatProvides(Composer\DependencyResolver\Pool $pool, $name, $bypassFilters = false)
```

### CODE REVIEW

File: Repository/AbstractAssetsRepository.php [hunk 1]
```diff
@@ -121,7 +121,7 @@ public function search($query, $mode = 0)
     /**
      * {@inheritdoc}
      */
-    public function whatProvides(Pool $pool, $name)
+    public function whatProvides(Pool $pool, $name, $bypassFilters = false)
```
Comment (francoispluchino) [inline Repository/AbstractAssetsRepository.php]: Can you remove this change, because it's already in the PR #205.

Comment (AnikanovD) [inline Repository/AbstractAssetsRepository.php]: This PR equivalent to #205, but also includes a fixed unit test.
Sorry if I do not understand something. This is my first pull request.

Comment (AnikanovD) [inline Repository/AbstractAssetsRepository.php]: @francoispluchino, this PR equivalent to #205, but also includes a fixed unit test.
Sorry if I do not understand something. This is my first pull request.

**Comment by francoispluchino** [comment]
Duplicate with 90ea5357652eba9bd8272cc374eda46ce6625df1 and #205.

**Comment by AnikanovD** [comment]
@francoispluchino, could you explain why merged #205 instead this pr, and create https://github.com/francoispluchino/composer-asset-plugin/commit/9ea76dd2cb1717a5645cf5b499598ed282db8f98, https://github.com/francoispluchino/composer-asset-plugin/commit/90ea5357652eba9bd8272cc374eda46ce6625df1?
i'm a newbie in the open source, and maybe don't understand guidelines.

**Comment by francoispluchino** [comment]
Because, it is preferable to specialize each PR, and fix problem by problem, or add feature by feature.

I hope I am clear enough :-).

**Comment by AnikanovD** [comment]
Clear enough :) Thanks for the reply.

**Pull request closed without merging**